src/main/scala/org/constellation/crypto/KeyUtils.scala

```diff
@@ -154,9 +161,25 @@ object KeyUtils {
       case None =>  bytes.map("%02x".format(_)).mkString
       case _ =>  bytes.map("%02x".format(_)).mkString(sep.get)
     }
-    // bytes.foreach(println)
   }
 
+  def publicKeyToHex(publicKey: PublicKey): String ={
+    val hex = bytes2hex(publicKey.getEncoded)
+    hex.slice(PublicKeyHexPrefixLength, hex.length)
+  }
+
+  def hexToPublicKey(hex: String): PublicKey = {
+    bytesToPublicKey(hex2bytes(PublicKeyHexPrefix + hex))
+  }
+
+  def privateKeyToHex(privateKey: PrivateKey): String ={
+    val hex = bytes2hex(privateKey.getEncoded)
+    hex.slice(PrivateKeyHexPrefixLength, hex.length)
+  }
+
+  def hexToPrivateKey(hex: String): PrivateKey = {
+    bytesToPrivateKey(hex2bytes(PrivateKeyHexPrefix + hex))
```

Comment:
I don't know the full scope yet of where which functions will be integrated and used but generally, since I see more utils functions introduced here, it's probably not a wasted attention to make sure it's in the back of peoples minds.
When e.g. private keys are dealt with, caution must be taken to e.g. not run routines over `Integer` or `hex` raw keys where it's not necessary, so that the system is exposed e.g. to [timing attacks](https://en.wikipedia.org/wiki/Timing_attack). Comparison or type casts, etc., that are implemented in a too efficient (normal) way may run quicker. (E.g. benchmarking a type cast of a raw private key that's secretly just `0x7777777` will reveal the range in which its in.) There are a million such things that need a good audit.

Reply:
Before I wrote the comment I looked a bit into where those calls are used till now and from that it seems to me that most of code relating to private keys is using all the library functions anyway - there's not all too much custom code yet and I hope the libraries are safe. 
Timing attacks in particular just from usage for signing may be far fetched anyway, but nontheless I've seen cryptographers discuss this very issue in projects.

Even Bitcoin had a [print-infinite-money](https://en.bitcoin.it/wiki/Value_overflow_incident) bug early on and Ethereum is half a fork of bitcoin and today most projects feel like EthereumGo forks. Since this is JVM, I'm just looking out to play devils advocate here if nobody is really dedicated in particular security beyond good practices.
Regarding revisiting later, I completely agree with that - but I also want to point out that those things which "already work" always get kicked down the role and if people start building software around it, these things run danger of never actually happening. With the key and hash stuff in particular, if one makers a possibly-later ticket for somewhen, one must be dedicated to not let "yeah but people working with the testnet in the latest years already use this and that public keys and so let's just let it be" be an excuse not to change it before  mainnet.

Reply:
For comparison:

https://github.com/bitcoin/bitcoin/blob/2c0867a1811bd7120aa66d98723ae96b065a3c1b/src/wallet/rpcdump.cpp#L697

https://github.com/bitcoin/bitcoin/blob/2c0867a1811bd7120aa66d98723ae96b065a3c1b/src/wallet/rpcdump.cpp#L745

https://github.com/bitcoin/bitcoin/blob/06ba77973e7f8938c8461bffeee3718a8f7712f5/src/key_io.cpp#L149

So we should probably have a memory cleanse step somewhere.